### PR TITLE
FIX: trim space in URL before executed

### LINKED ISSUE #788: The Url should be trimmed before being executed
**Describe the bug**
If a space is present in front of the URL GE will query `https://%20graph.microsoft.com/v1.0/me` instead of `https://graph.microsoft.com/v1.0/me`.

Inadvertently adding a space is common when you copy/paste things.

**To Reproduce**
1. Open GE
2. Add a space in front of the URL to query i.e ` https://graph.microsoft.com/v1.0/me` (note the space in front)
3. Run the query
4. the request fails

**Expected behavior**
GE to trim spaces on the URL so the request succeeds. 
[AB#7480](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7480)

### PR DESCRIPTION
**Pull request by ezkemboi**
- Fixes #788

## Overview

- GE to trim spaces on the URL so the request succeeds.

### Demo

Screenshot 2021-02-02 185016

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* How to test this PR
* Prefer bulleted description
* Start after checking out this branch
* Include any setup required, such as bundling scripts, restarting services, etc.
* Include test case, and expected output

### CODE REVIEW

**Comment by jobala** [comment]
Instead of running this magically should we instead have a url validation error message?

**Comment by ezkemboi** [comment]
@baywet any comment on @jobala suggestion?

File: src/app/views/query-runner/QueryRunner.tsx [hunk 1]
```diff
@@ -87,6 +87,8 @@ export class QueryRunner extends Component<
     }
 
     if (actions) {
+      // remove whitespaces
+      sampleQuery.sampleUrl = sampleQuery.sampleUrl.replace(/\s+/g, '');
```
Comment (baywet) [inline src/app/views/query-runner/QueryRunner.tsx:91]:
```suggestion
      sampleQuery.sampleUrl = sampleQuery.sampleUrl.trim();
```

**Review by baywet — CHANGES_REQUESTED** [review]
The current implement would replace spaces within the URL, which is not valid, for example some odata query parameters can contain spaces and that'd be valid. I think we should only consider the spaces at the end and the beginning so the request isn't changed and doesn't fail because of a typo

**Comment by ezkemboi** [comment]
Changes updated accordingly.

**Review by baywet — APPROVED** [review]

**Comment by jobala** [comment]
@ezkemboi @baywet trimming the whitespaces doesn't solve this issue. Odata query parameters come after the `?`. @ezkemboi can you implement a URL validator for the path before a querystring?

**Review by jobala — CHANGES_REQUESTED** [review]
@ezkemboi @baywet trimming the whitespaces doesn't solve this issue. Odata query parameters come after the ?. @ezkemboi can you implement a URL validator for the path before a querystring?

**Comment by ezkemboi** [comment]
@jobala, I will work on the changes.

**Comment by baywet** [comment]
@jobala Thanks for double checking. I was looking into the nature of the change without knowing the code base very well but I didn't test the change myself.

**Comment by ezkemboi** [comment]
@jobala, I have been busy a little, but I will try to work on the changes requested over this weekend.

**Comment by ElinorW** [comment]
> @jobala, I have been busy a little, but I will try to work on the changes requested over this weekend.

Hey @ezkemboi any progress on this?

**Comment by ezkemboi** [comment]
@ElinorW  I did not manage to work on this weekend, I had an interview this Monday and I was preparing for it over the weekend.

**Comment by ddyett** [comment]
@ezkemboi are you still planning to work on this?

**Comment by thewahome** [comment]
Thanks so much @ezkemboi for contributing to this. 

It has been open for a long time and taking advantage of the new code changes, I am closing this to work on it based on the below quoted suggestion

> Instead of running this magically should we instead have a url validation error message?

I will rope you in to see how it goes.

**Comment by ezkemboi** [comment]
I will double-check on this PR and its request and see if there is the best way, I can get a new PR out of it based on @jobala requests and comments highlighted previously.

**Pull request closed without merging**